cmd/juju/application/deployer/bundle.go

```diff
@@ -173,6 +178,28 @@ Please repeat the deploy command with the --trust argument if you consent to tru
 	return nil
 }
 
+// checkExplicitBase returns an error if the image-id constraint is used and
+// there is no base (series) explicitly defined by the user.
+func (d *deployBundle) checkExplicitBase(bundleData *charm.BundleData) error {
+	for _, applicationSpec := range bundleData.Applications {
+		charmURL, err := resolveCharmURL(applicationSpec.Charm, d.defaultCharmSchema)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		cons, err := constraints.Parse(applicationSpec.Constraints)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if (cons.HasImageID() || d.modelConstraints.HasImageID()) &&
+			applicationSpec.Series == "" &&
+			charmURL.Series == "" &&
+			bundleData.Series == "" {
+			return errors.Forbiddenf("base must be explicitly provided when image-id constraint is used")
```

Comment:
I wonder if this will be confusing as "base" is not allowed in the bundle specification today. Nor can you use `--base` on the cli for bundles.  Series is old style, but makes more sense when telling a user how to correct the error?

Reply:
yeah, indeed. I rename the method and changed the returned error to `series must be explicitly...`